subworkflows/local/contaminant_filter.nf

```diff
@@ -81,9 +112,30 @@ workflow CONTAMINANT_FILTER {
     cdna_reads.set { ncrna_reads }
 
     if (params.ncrna) {
-        BLAT_NCRNA ( Channel.value( 'ncrna' ), ch_reference_hairpin, ch_ncrna )
-        ch_versions = ch_versions.mix(BLAT_NCRNA.out.versions)
-        INDEX_NCRNA ( BLAT_NCRNA.out.filtered_set )
+        // Search which hairpin miRNAs are present in the ncRNA data
+        BLAT_NCRNA(ch_reference_hairpin, ch_ncrna)
+
+        // Extract the significant hits
+        ch_program = Channel.of('BEGIN{FS="\t"}{if(\$11 < 1e-5) print \$2;}').collectFile(name:"program.txt")
```

Comment:
Placeholder for removing ch_program if definining it on top of the workflow outside of the if statements